components/forms/CreateTxForm/MsgForm/MsgCreateVestingAccountForm.tsx

```diff
@@ -0,0 +1,190 @@
+*/
+const getMinEndTime = (): string => {
+  const minTimestamp = Date.now() + 1000 * 60 * 60 * 24 * 30;
+  const minDate = new Date(minTimestamp);
+
+  const minMonth = minDate.getMonth() + 1; // It's 0-indexed
+  const minMonthStr = minMonth < 10 ? `0${minMonth}` : String(minMonth);
+
+  const minDay = minDate.getDate();
+  const minDayStr = minDay < 10 ? `0${minDay}` : String(minDay);
+
+  const minHours = minDate.getHours();
+  const minHoursStr = minHours < 10 ? `0${minHours}` : String(minHours);
+
+  const minMinutes = minDate.getMinutes();
+  const minMinutesStr = minMinutes < 10 ? `0${minMinutes}` : String(minMinutes);
+
+  return `${minDate.getFullYear()}-${minMonthStr}-${minDayStr}T${minHoursStr}:${minMinutesStr}`;
+};
+
+interface MsgCreateVestingAccountFormProps {
+  readonly fromAddress: string;
+  readonly setMsgGetter: (msgGetter: MsgGetter) => void;
+  readonly deleteMsg: () => void;
+}
+
+const MsgCreateVestingAccountForm = ({
+  fromAddress,
+  setMsgGetter,
+  deleteMsg,
+}: MsgCreateVestingAccountFormProps) => {
+  const { state } = useAppContext();
+  assert(state.chain.addressPrefix, "addressPrefix missing");
+
+  const [toAddress, setToAddress] = useState("");
+  const [amount, setAmount] = useState("0");
+  const minEndTime = getMinEndTime();
+  const [endTime, setEndTime] = useState(minEndTime);
+  const [delayed, setDelayed] = useState(true);
+
+  const [toAddressError, setToAddressError] = useState("");
+  const [amountError, setAmountError] = useState("");
+  const [endTimeError, setEndTimeError] = useState("");
+
+  useEffect(() => {
+    try {
+      assert(state.chain.denom, "denom missing");
+
+      setToAddressError("");
+      setAmountError("");
+      setEndTimeError("");
+
+      const isMsgValid = (msg: TxMsg): msg is TxMsgCreateVestingAccount => {
+        assert(state.chain.addressPrefix, "addressPrefix missing");
+
+        const addressErrorMsg = checkAddress(toAddress, state.chain.addressPrefix);
+        if (addressErrorMsg) {
+          setToAddressError(
+            `Invalid address for network ${state.chain.chainId}: ${addressErrorMsg}`,
+          );
+          return false;
+        }
+
+        if (!amount || Number(amount) <= 0) {
+          setAmountError("Amount must be greater than 0");
+          return false;
+        }
+
+        if (!endTime) {
+          setEndTimeError("End time is required");
+          return false;
+        }
+
+        return isTxMsgCreateVestingAccount(msg);
+      };
+
+      const amountInAtomics = amount
+        ? Decimal.fromUserInput(amount, Number(state.chain.displayDenomExponent)).atomics
+        : "0";
+
+      const msg: TxMsgCreateVestingAccount = {
+        typeUrl: "/cosmos.vesting.v1beta1.MsgCreateVestingAccount",
+        value: {
+          fromAddress,
+          toAddress,
+          amount: [{ amount: amountInAtomics, denom: state.chain.denom }],
+          endTime: timestampFromDatetimeLocal(endTime),
+          delayed,
```

Comment:
Could you pull this out into a `const createMsgValue: MsgCreateVestingAccount = { ...` to ensure we have type-safety for that part?

Reply:
oh, you have that in TxMsgCreateVestingAccount. Never mind then.